src/node.rs

```diff
@@ -0,0 +1,223 @@
+use crate::message::{FindValueResult, Message};
+use crate::routing_table::RoutingTable;
+use crate::utils::NodeId;
+use log::{debug, error, info, warn};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::net::{ UdpSocket};
+use std::net::SocketAddr;
+
+use tokio::sync::mpsc;
+use tokio::time::{interval, Duration};
+
+use crate::utils::{ALPHA, BOOTSTRAP_NODES, K};
+use bincode::{deserialize, serialize};
+
+pub struct KademliaNode {
+    id: NodeId,
+    addr: SocketAddr,
+    routing_table: RoutingTable,
+    storage: HashMap<Vec<u8>, Vec<u8>>,
+    socket: Arc<UdpSocket>,
+    shutdown: mpsc::Receiver<()>,
+}
+
+impl KademliaNode {
+    pub async fn new(addr: SocketAddr) -> std::io::Result<(Self, mpsc::Sender<()>)> {
+        let id = NodeId::new();
+        let socket = UdpSocket::bind(addr).await?;
+        let (shutdown_sender, shutdown_receiver) = mpsc::channel(1);
```

Comment:
semantically, if we're thinking that something _outside_ of this `node` mod would be sending to the channel to send shutdown signals, we might want to instantiate this channel _outside_ of this `node` mod, say in `src/main.rs`.